### PR TITLE
Node.js update query parameters

### LINKED ISSUE #1503: Nodejs: split query parameters into multiple structs
https://github.com/iotaledger/iota-sdk/pull/1428

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Node.js update query parameters

## Links to any relevant issues

Closes https://github.com/iotaledger/iota-sdk/issues/1503

## Type of change

- Breaking change (fix or feature that would cause existing functionality to not work as expected)

## How the change has been tested

tsc

### CODE REVIEW

File: bindings/nodejs/lib/types/client/query-parameters.ts [hunk 1]
```diff
@@ -1,188 +1,162 @@
...
+    /// Filters outputs based on the presence of timelock unlock condition.
+    hasTimelock?: boolean;
+    /// Returns outputs that are timelocked before a certain slot index.
+    timelockedBefore?: SlotIndex;
+    /// Returns outputs that are timelocked after a certain slot index.
+    timelockedAfter?: SlotIndex;
+    /// Filters outputs based on the presence of validated Sender (bech32 encoded).
+    sender?: Bech32Address;
+    /// Filters outputs based on matching Tag Block.
+    tag?: String;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/client/query-parameters.ts]: unhappy CI

File: bindings/nodejs/lib/types/client/query-parameters.ts [hunk 2]
```diff
@@ -1,188 +1,162 @@
...
+    /// Returns outputs that are timelocked before a certain slot index.
+    timelockedBefore?: SlotIndex;
+    /// Returns outputs that are timelocked after a certain slot index.
+    timelockedAfter?: SlotIndex;
+    /// Filters outputs based on bech32-encoded issuer address.
+    issuer?: Bech32Address;
+    /// Filters outputs based on the presence of validated Sender (bech32 encoded).
+    sender?: Bech32Address;
+    /// Filters outputs based on matching Tag Block.
+    tag?: String;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/client/query-parameters.ts]: Unhappy CI

File: bindings/nodejs/lib/types/client/query-parameters.ts [hunk 3]
```diff
@@ -1,188 +1,162 @@
...
-/** Filter foundry outputs based on bech32-encoded address of the controlling account. */
-interface AccountAddress {
-    accountAddress: Bech32Address;
+export interface CommonOutputQueryParameters {
+    /// The maximum amount of items returned in one call. If there are more items, a cursor to the next page is
+    /// returned too. The parameter is ignored when pageSize is defined via the cursor parameter.
+    pageSize?: number;
+    /// Starts the search from the cursor (createdSlotIndex+outputId.pageSize). If an empty String is provided, only
+    /// the first page is returned.
+    cursor?: String;
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/client/query-parameters.ts]: Unhappy CI

File: bindings/nodejs/lib/types/client/query-parameters.ts [hunk 4]
```diff
@@ -1,188 +1,163 @@
 // Copyright 2021-2023 IOTA Stiftung
-// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier?: Apache-2.0
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/client/query-parameters.ts]: :trollface:

File: bindings/nodejs/lib/types/client/bridge/client.ts [hunk 5]
```diff
@@ -365,3 +281,88 @@ export type __CallPluginRouteMethod__ = {
         request?: string;
     };
 };
+
+// inx-indexer methods
+export interface __GetOutputIdsMethod__ {
```
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/client/bridge/client.ts:287]:
```suggestion

export interface __GetOutputIdsMethod__ {
```

File: bindings/nodejs/lib/types/client/query-parameters.ts (hunk 4 above)
Comment (thibault-martinez) [inline bindings/nodejs/lib/types/client/query-parameters.ts]:
```suggestion
// SPDX-License-Identifier: Apache-2.0
```

**Review by DaughterOfMars — COMMENTED** [review]
much better

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

File: bindings/nodejs/lib/types/client/query-parameters.ts (hunk 4 above)
Comment (Thoralf-M) [reply bindings/nodejs/lib/types/client/query-parameters.ts]: :trollface:

**Pull request merged**